polaris/hub/client.py

```diff
@@ -347,10 +371,14 @@ def get_benchmark(
             else MultiTaskBenchmarkSpecification
         )
 
-        if not verify_checksum:
-            response.pop("md5Sum", None)
+        benchmark = benchmark_cls(**response)
 
-        return benchmark_cls(**response)
+        if should_verify_checksum(verify_checksum, benchmark.dataset):
+            benchmark.verify_checksum()
+        else:
+            benchmark.md5sum = response["md5Sum"]
```

Comment:
Is there ever a case where we get a `md5Sum` in the response but don't want to use it as the artifact's checksum? Just wondering if this could possibly be simplified to assume `benchmark.md5sum` is either `response["md5Sum"]` or the result of calling `verify_checksum()`

Reply:
> could possibly be simplified to assume benchmark.md5sum is either response["md5Sum"] or the result of calling verify_checksum()

I may be missing something here, but I think that's the assumption right now! How would you simplify it?

Reply:
I was thinking checking for the presence of the response `"md5Sum"` could replace the `should_verify_checksum` call in this case, something like:
```
if response["md5sum"]:
    benchmark.md5sum = response["md5Sum"]
else:
    benchmark.verify_checksum()
```

(but haven't tested that, maybe it won't work.. was just wondering if we really need `should_verify_checksum`)..

Reply:
Interesting idea, but I don't that will work. Even if we download the right checksum, verifying it is still needed to make sure we pulled in the rest of the data correctly as well.